### DISCUSSION TITLE
How should the current "Preferred Storage Location" be represented in the UI.

### DISCUSSION THREAD

**Discussion by jmchilton** [galaxyproject/galaxy#18163] [General Discussion]
@itisAliRH pointed out while reviewing user defined object stores (which didn't introduce object store selection but uses it heavily)

> Preferred storage location is not visible to the user until hovering on the icon (I'd put the selected name next to its icon)

<img width="613" alt="Screenshot 2024-05-16 at 11 58 05 AM" src="https://github.com/galaxyproject/galaxy/assets/216771/a5dee2cd-376c-4f1f-8b41-1af1243a22be">

My gut says that this feature is still pretty niche - even with something as easy as scratch storage on main - and so devoting more screen real estate to it is a mistake.

A more pressing issue is perhaps creating a visual indicator that makes it clear this only applies to new datasets. Everyone who has talked to me about this concept has pointed out it is unclear what happens to existing datasets. I've tried to make it clear with text help but the problem persists. Possibly because it is the only prospective option displayed in the history panel - datasets, quota, metadata are all established things? Any ideas here?

I think of the four ways to set a target storage location (user default, tool run, workflow run, and history-specific default) - history provides the least utility and (as @martenson points out - https://github.com/galaxyproject/galaxy/issues/18128#issuecomment-2115663363) is the most visible. Maybe just hiding it in advanced settings related to the history would help the status quo?

**Comment by davelopez** [comment]
> Maybe just hiding it in advanced settings related to the history would help the status quo?

FWIW the Storage Dashboard seems like the natural place for these kinds of settings and offers a dedicated place to direct users to. It also has plenty of screen space to inform and educate on how it works and how to use it.
